wrangler-service/src/main/java/io/cdap/wrangler/service/adls/ADLSHandler.java

```diff
@@ -0,0 +1,382 @@
+
+    /**
+     * Lists ADLS directory's contents for the given prefix path.
+     *
+     * @param request   HTTP Request handler.
+     * @param responder HTTP Response handler.
+     */
+    @TransactionPolicy(value = TransactionControl.EXPLICIT)
+    @ReadOnly
+    @GET
+    @Path("contexts/{context}/connections/{connection-id}/adls/explore")
+    public void listADLSDirectory(HttpServiceRequest request, HttpServiceResponder responder,
+                                  @PathParam("context") String namespace,
+                                  @PathParam("connection-id") final String connectionId,
+                                  @Nullable @QueryParam("path") final String path) {
+        respond(request, responder, namespace, ns -> {
+            try {
+                Connection connection = getValidatedConnection(new NamespacedId(ns, connectionId), ConnectionType.ADLS);
+                final String defaultPath = "/";
+                ADLStoreClient adlStoreClient = initializeAndGetADLSClient(connection);
+                List<ADLSDirectoryEntryInfo> adlsDirectoryEntryInfos;
+                if (path == null || path.equals("")) {
+                    adlsDirectoryEntryInfos = ADLSUtility.initClientReturnResponse(adlStoreClient, defaultPath);
+                } else {
+                    adlsDirectoryEntryInfos = ADLSUtility.initClientReturnResponse(adlStoreClient, path);
+                }
+                return new ServiceResponse<>(adlsDirectoryEntryInfos);
+            } catch (ADLException e) {
+                throw new StatusCodeException(e.getMessage(), e, e.httpResponseCode);
+            }
+        });
+    }
+
+    /**
+     * Reads ADLS file into workspace
+     *
+     * @param request   HTTP Request handler.
+     * @param responder HTTP Response handler.
+     */
+    @POST
+    @ReadWrite
+    @Path("contexts/{context}/connections/{connection-id}/adls/read")
+    public void loadADLSFile(HttpServiceRequest request, HttpServiceResponder responder,
+                             @PathParam("context") String namespace,
+                             @PathParam("connection-id") String connectionId,
+                             @QueryParam("path") String filePath, @QueryParam("lines") int lines,
+                             @QueryParam("sampler") String sampler, @QueryParam("fraction") double fraction,
+                             @QueryParam("scope") String scope) {
+        if (Strings.isNullOrEmpty(scope)) {
+            scope = WorkspaceDataset.DEFAULT_SCOPE;
+        }
+        final String scopeS = scope;
+        respond(request, responder, namespace, ns -> {
+            try {
+                if (Strings.isNullOrEmpty(connectionId)) {
+                    throw new BadRequestException("Required path param 'connection-id' is missing in the input");
+                }
+                if (lines == 0) {
+                    throw new NumberFormatException("lines to extract only limit random lines should not be zero");
```

Comment:
this should be a BadRequestException, as it is invalid user input and should result in a 400 response code.

Reply:
This is a confusing error message to me, something like "The maximum number of sampled lines cannot be zero." seems more clear.

Reply:
Agreed, have changed the exception and its message.